triggers/service/auth/src/main/scala/com/daml/oauth/middleware/Config.scala

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.oauth.middleware
+
+import akka.http.scaladsl.model.Uri
+import com.daml.ports.Port
+
+private[middleware] case class Config(
+    // Port the middleware listens on
+    port: Port,
+    // Uri of the OAuth2 server
+    oauthUri: Uri,
+    // OAuth2 client properties
+    clientId: String,
+    clientSecret: String,
+)
+
+private[middleware] object Config {
+  private val Empty =
+    Config(
+      port = Port.Dynamic,
+      oauthUri = Uri().withScheme("http"),
```

Comment:
It allows to implement the `--oauth-host` and `--oauth-port` flags concisely. Those flags are also marked as required, so we won't end up with just `http://` from the command-line flags. Is there a better way to do that?

Reply:
Is there a reason why we accept `oauth-host` and `oauth-port` instead of accepting a single parameter that specifies a URI? Host and port makes sense for gRPC but for HTTP Requests it seems a bit weird.

Reply:
Fair enough, no good reason, really. I'll make it a single flag.